plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -241,163 +246,57 @@ void deleteBlobsByPrefix(String prefix) throws IOException {
      * @param blobNames names of the bucket to delete
      */
     void deleteBlobs(Collection<String> blobNames) throws IOException {
-        if (blobNames == null || blobNames.isEmpty()) {
+        if ((blobNames == null) || blobNames.isEmpty()) {
             return;
         }
-
-        if (blobNames.size() == 1) {
-            deleteBlob(blobNames.iterator().next());
+        if (blobNames.size() < 5) {
+            for (final String blobName : blobNames) {
+                deleteBlob(blobName);
+            }
```

Comment:
This "optimization" is not justifiable per se, but it is required to get around another URL building blunder inside the google cloud library. [Batched requests don't honor the hostname + port configuration](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/52b727aef88ae76984aa3c02b4d7067e198d34b7/google-cloud-storage/src/main/java/com/google/cloud/storage/spi/v1/HttpStorageRpc.java#L191). All batched request paths are hardcoded to `www.googleapis.com` 😢 
REST integ tests don't delete more than 3 files in a single batch, so this code branch is required to make the integ tests work.